site/src/pages/DeploySettingsPage/OAuth2AppsSettingsPage/EditOAuth2AppPage.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import {
+  deleteOAuth2App,
+  deleteOAuth2AppSecret,
+  postOAuth2AppSecret,
+  putOAuth2App,
+} from "api/api";
+import type * as TypesGen from "api/typesGenerated";
+import {
+  oauth2App,
+  oauth2AppSecrets,
+  oauth2AppSecretsKey,
+} from "api/queries/oauth2";
+import { displayError, displaySuccess } from "components/GlobalSnackbar/utils";
+import { FC, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { EditOAuth2AppPageView } from "./EditOAuth2AppPageView";
+import { pageTitle } from "utils/page";
+import { Helmet } from "react-helmet-async";
+
+const EditOAuth2AppPage: FC = () => {
+  const navigate = useNavigate();
+  const { appId } = useParams() as { appId: string };
+  const queryClient = useQueryClient();
+  const [newAppSecret, setNewAppSecret] = useState<
+    TypesGen.OAuth2AppSecretFull | undefined
+  >(undefined);
+
+  const oauth2AppQuery = useQuery(oauth2App(appId));
+  const appName = oauth2AppQuery.data?.name;
+
+  const deleteMutation = useMutation({
+    mutationFn: deleteOAuth2App,
+    onSuccess: async () => {
+      displaySuccess(
+        `You have successfully deleted the OAuth2 application "${appName}"`,
```

Comment:
I omitted the invalidations on all the navigations but I should add them.  I remember there being talk about caching, maybe in those cases the invalidations are required even with navigation?

I decided against a check because I think it is not possible to have an undefined app name here because with no app name there would be no app and the page would still be loading, so no delete button to make a delete request.

That said, I think this is a bad way to do it and maybe this guarantee will not always hold.  According to our conversation above I am going to move these to `queries` and then in the callback I will do:

```
await deleteMutation.mutate(app.id)
displaySuccess(`bla bla ${app.name}`)
```

Which I think will sort it!